### PR TITLE
LPS-134450 Semver

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 1]
```diff
@@ -892,27 +909,47 @@ private void _populateWithTasks(Document document, Instance instance) {
...
 
-				for (Object assigneeId : (List<?>)task.get("assigneeIds")) {
+				List<Long> roleIds;
+
+				if (task.get("assigneeGroupIds") != null) {
+					assigneeGroupIds = ListUtil.toList(
+						(List<?>)task.get("assigneeGroupIds"),
+						GetterUtil::getLong);
+				}
+
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:940]: @seltong, please, define the `assigneeIds` variable below this condition, thanks!

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 2]
```diff
@@ -892,27 +909,47 @@ private void _populateWithTasks(Document document, Instance instance) {
...
+				if (task.get("assigneeIds") != null) {
+					assigneeIds = ListUtil.toList(
+						(List<?>)task.get("assigneeIds"), GetterUtil::getLong);
+				}
+
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:945]: @seltong, please, define the `mapGroupRoleIds` variable below this condition, thanks!

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 3]
```diff
@@ -892,27 +909,47 @@ private void _populateWithTasks(Document document, Instance instance) {
...
+				for (long groupId : assigneeGroupIds) {
+					mapGroupRoleIds.put(groupId, _getRoleIdsByGroupId(groupId));
+				}
+
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:949]: @seltong, please, define the `reviewer` variable below this loop, thanks!

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 4]
```diff
@@ -921,6 +958,16 @@ else if (Objects.equals(
 					}
 				}
 
+				for (int i = 0; i < assigneeGroupIds.size(); i++) {
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:961]: @seltong, please, define the `roleIds` variable above this loop, thanks!

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 5]
```diff
@@ -921,6 +958,16 @@ else if (Objects.equals(
...
+					roleIds = mapGroupRoleIds.get(assigneeGroupIds.get(i));
+
+					if (roleIds.contains(assigneeIds.get(i))) {
+						reviewer = true;
+
+						break;
+					}
+				}
+
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:970]: Hello @rafaprax, we can create a new method with this changes to facilitate other developer understanding?, for example, `isReviewer`

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/InstanceResourceTest.java [hunk 6]
```diff
@@ -421,6 +444,41 @@ protected Instance testGetProcessInstancesPage_addInstance(
 		return instance;
 	}
 
+	protected Instance testGetProcessInstancesPage_addInstance(
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/InstanceResourceTest.java:447]: @seltong, this method have some contents duplicated in this [other method](https://github.com/liferay-workflow/liferay-portal/pull/452/commits/fee14b6bfd75004b500afc4ee56ad44d0a0f0458#diff-a74bcccf83965baf2e916a5d957a8244f06e9973aa2e5f66224950a882b875b1R419), could you join both?

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/InstanceResourceTest.java [hunk 7]
```diff
@@ -181,6 +197,13 @@ public void testGetProcessInstancesPage() throws Exception {
 			instances -> assertEqualsIgnoringOrder(
 				Arrays.asList(instance1, instance2), instances));
 
+		_testGetProcessInstancesPage(
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/InstanceResourceTest.java:200]: @seltong, could you add the case when exists two sites but in just one the user will have permission to reviewer?

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/model/listener/KaleoTaskInstanceTokenModelListener.java [hunk 8]
```diff
@@ -64,21 +65,14 @@ public void onAfterCreate(KaleoTaskInstanceToken kaleoTaskInstanceToken)
...
-					() -> null
-				);
+				Long[] assigneeIds = _getAssigneeInfo(
+					kaleoTaskAssignmentInstances,
+					KaleoTaskAssignmentInstance::getAssigneeClassPK);
+
+				Long[] assigneeGroupIds = _getAssigneeInfo(
+					kaleoTaskAssignmentInstances,
+					KaleoTaskAssignmentInstance::getGroupId);
+
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/model/listener/KaleoTaskInstanceTokenModelListener.java:75]: @seltong, please could you do this changes above in [this commit](https://github.com/liferay-workflow/liferay-portal/pull/452/commits/5c3150ed95043063e63304b8365e99828a912692)?, thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java (hunk 5 above)
Comment (rafaprax) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:970]: Yeap. We can.

**Comment by rafaprax** [comment]
Hey @MariMalta and @seltong please take a look at the broken tests. The failures seem related to your changes. Thx

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Just started reviewing :)

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:stop

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
Just started reviewing :)

**Comment by rafaprax** [comment]
@seltong will resend this.

**Pull request closed without merging**